pghoard/basebackup.py

```diff
@@ -710,7 +710,12 @@ def create_and_upload_chunks(
                     # and this assumption greatly simplifies the logic.
                     task_to_wait = pending_compress_and_encrypt_tasks.pop(0)
                     chunk_files.append(task_to_wait.result())
-
+                    if len(chunks) > 0:
```

Comment:
This is redundant, it's not possible for chunks to be empty here. Same for the other check. Otherwise looks good to me. Apparently there's some issue with tests currently, though, as they're failing here as well as some other pull requests. (Unrelated to this but would be nice to have passing tests before merging anything.)